libs/javalib/worker/src/mill/javalib/zinc/ZincWorker.scala

```diff
@@ -106,9 +106,16 @@ class ZincWorker(
           loaderLibraryOnly = ClasspathUtil.rootLoader,
           libraryJars = Array(libraryJarNameGrep(
             compilerClasspath,
-            // we don't support too outdated dotty versions
-            // and because there will be no scala 2.14, so hardcode "2.13." here is acceptable
-            if (JvmWorkerUtil.isDottyOrScala3(key.scalaVersion)) "2.13." else key.scalaVersion
+            if (
+              JvmWorkerUtil.isDottyOrScala3(
+                key.scalaVersion
+              ) && !JvmWorkerUtil.isUnifiedLibraryAvailable(key.scalaVersion)
+            )
+              // we don't support too outdated dotty versions
+              // and because there will be no scala 2.14, so hardcode "2.13." here is acceptable.
+              // Since 3.8, we can just use the unified library.
+              "2.13."
+            else key.scalaVersion
```

Comment:
I typically don't comment on formatting, but this is too hard to read.
Took the opportunity to revise the comments.

```suggestion
            if (
              JvmWorkerUtil.isDottyOrScala3(key.scalaVersion) && 
              !JvmWorkerUtil.isUnifiedLibraryAvailable(key.scalaVersion)
            )
              // All stable Scala 3 versions before 3.8 use the Scala 2.13 library
              "2.13."
            else
              // Since Scala 3.8, we can just use the unified library
              key.scalaVersion
```